frontend_react/src/api/SocketService.ts

```diff
@@ -0,0 +1,46 @@
+import { SocketServiceInterface } from "../utils/SocketServiceInterface";
+
+const SocketService: SocketServiceInterface = {
+  webSocket: undefined,
+  connect: (websocketAddress: string) => {},
+  disconnect: () => {},
+};
+
+SocketService.connect = (websocketAddress) => {
+  if (SocketService.webSocket === undefined) {
+    SocketService.webSocket = new WebSocket(websocketAddress);
```

Comment:
I was just curious where we are importing `WebSocket` from. I'm a little rusty on socket.io, is this a socket.io function to create a brand new socket? I didn't see any references to socket.io other than the package.json so I was a little confused.

Reply:
Good catch @ParsaHonarmand. I did not end up using socket.io package because it requires a node.js server, and whatever server MindFuel is using on their backend, it didn't work with it. Hence the base WebSocket implementation. I'll remove the socket.io from package.json.